designs/virtualization/4.x/expose-guest-data/expose guest data.md

```diff
@@ -0,0 +1,53 @@
+# Expose Guest Agent Data
+
+When the Guest Agent is installed on a VM some additional valuable data can be collected then surfaced in the UI.
+When the Guest Agent isn’t installed we will inform the users about that.
+
+We can report that in the VM’s Overview tab under the Status card and provide a message: “This virtual machine does not have a Guest Agent installed. Some metrics, and management features will not be available or accurate” and add an external link to view the details about the installation procedure of the Guest Agent, if the users are interested to do so. The link will lead the users to the instructions on how to install the Guest Agent. Linux users will need to download and install the Guest Agent online, while Windows users will get the installation on a CD ROM.
```

Comment:
accurate should be inaccurate :)

Reply:
Do we have an example of something that might not be accurate? It seems a little odd to show information that we know ahead of time could be misleading or incorrect... should we even bother showing those things at all?

Reply:
@andybraren Hi,

> Do we have an example of something that might not be accurate? 

For example the "Operating System" we show in overview and inventory:
   a - without GA - a label user created to indicate what OS they intend to install on the VM.
   b - with GA - the actual OS installed.
In this case we may show wrong information e.g. "windows" without GA because admin intended to install windows, but "RHEL 8" with GA because this is the OS that was actually installed, or partial information, e.g. "fedora" without GA and "Fedora 30 (Cloud Edition)" with GA.

Reply:
> should we even bother showing those things at all?

Sometimes :-) , in the OS example ( https://github.com/openshift/openshift-origin-design/pull/380#discussion_r404545463 ) the two parts of the information may be important:
a - If actual OS installed is different than the one the admin intended, it's important to know about it, it may indicate a problem.
b - if actual  OS installed is the same as intended, than original data may be redundant, because with GA we will get the same data but with more information.

Reply:
ya I agree, if they've chosen Windows in the install flow (for example) and don't have the GA installed it makes sense to show them what we can which would be what they chose. Ideally they install the GA and we can provide the most accurate info but in most cases I'd bet the one they choose will be accurate anyway.

Reply:
Sounds good, thank you for the example @yaacov!